File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
// first seperate all the tokens using / as tokenizer
//         then check if its empty continue , . then also continue, and if ".."then pop the top element from stack
//         return 
      
      String[] paths = path.split("/");
        Stack<String> st = new Stack<>();
        for(String s:paths){
            if(s.equals("")) continue;
            else if(s.equals(".")) continue;
            else if(s.equals("..")) 
                if(!st.isEmpty()) st.pop();
                else continue;
            else st.add(s);
        }
        return "/"+String.join("/", st);
    }
}

File: 0071-simplify-path/0071-simplify-path.cpp
class Solution {
public:
    string simplifyPath(string s) {
         int n = s.length();
        stack<string> st;
        for(int i=0; i<n; i++){
            if(s[i]=='/') continue;
            string tmp = "";
            while(i<n && s[i]!='/') tmp+=s[i++];
            if(tmp==".") continue;
            if(tmp==".."){
                if(!st.empty()) st.pop();
                continue;
            }
            else st.push(tmp);
        }
        
        string p = "";
        
        while(!st.empty()){
            p ="/"+st.top()+p;
            st.pop();
        }
        if(p=="")return"/";
        return p;
    }
};